lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -8452,14 +8494,11 @@ bool SwiftASTContext::GetCompileUnitImportsImpl(
   if (cu_imports.size() == 0)
     return true;
 
-  Progress progress(
-      llvm::formatv("Getting Swift compile unit imports for '{0}'",
-                    compile_unit->GetPrimaryFile().GetFilename()),
-      cu_imports.size());
-
+  Progress progress("Importing module used by compile unit");
   size_t completion = 0;
   for (const SourceModule &module : cu_imports) {
-    progress.Increment(++completion);
+    progress.Increment(
+        ++completion, compile_unit->GetPrimaryFile().GetFilename().AsCString());
```

Comment:
I think this is wrong: You just moved the use of `compile_unit` in the loop to use it in the message detail but it never changes. I think you should leave the progress message the same it was and use the `SourceModule &module` variable to give extra information in the detailed messaged.

Reply:
Ah, I didn't notice that `compile_unit` was the wrong variable to use here. @adrian-prantl made the suggestion to say "Importing module used by compile unit..." but maybe we could say something like "Importing module used by compile unit <compile unit name>: <module>" with the module as the detail but that might be overkill

Reply:
Why would that be overkill ?